src/Symfony/Component/Config/Definition/ArrayNode.php

```diff
@@ -209,8 +209,8 @@ public function getDefaultValue()
      */
     public function addChild(NodeInterface $node)
     {
-        $name = $node->getName();
-        if (!strlen($name)) {
+        $name = (string) $node->getName();
```

Comment:
Tried to add validation into base node if string provided as name, UTs crashed.

Original code allows 0 and false, so I just put this visible - inspire of PhpDoc it's possible to provide non-string values as name, so perhaps it's a new bug discovered.